Remotion/Data/DomainObjects/Infrastructure/ObjectIDStringSerialization/ObjectIDStringSerializer.cs

```diff
@@ -123,14 +123,30 @@ private ObjectID ParseWithCustomErrorHandler (string objectIDString, Func<string
       var valuePart = objectIDString.Substring(indexOfValuePart, indexOfTypeDelimiter - indexOfValuePart);
       var typePart = objectIDString.Substring(indexOfTypePart);
 
-      Type? type = TypeUtility.GetType(typePart, throwOnError: false);
-      if (type == null)
+#if NET
+      IObjectIDValueParser? valueParser = typePart switch
       {
-        return errorHandler(
-            string.Format("Serialized ObjectID '{0}' is invalid: '{1}' is not the name of a loadable type.", objectIDString, typePart));
+          "System.Guid" => GuidObjectIDValueParser.Instance,
+          "System.Int32" => Int32ObjectIDValueParser.Instance,
+          "System.String" => StringObjectIDValueParser.Instance,
+          _ => null
+      };
+#else
+      IObjectIDValueParser? valueParser = null;
+      var typeParts = typePart.Split(new[] { ',' }, 3);
+      var isFromMscorlib = typeParts.Length < 2 || typeParts[1].Trim() == "mscorlib";
```

Comment:
I so want C# list patterns :D

Reply:
C# 11: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/patterns#list-patterns

Reply:
Why I mentioned it :D